app/assets/javascripts/discourse/initializers/topic-footer-buttons.js.es6

```diff
@@ -0,0 +1,116 @@
+import { withPluginApi } from "discourse/lib/plugin-api";
+
+export default {
+  name: "topic-footer-actions",
+
+  initialize() {
+    withPluginApi("0.8", api => {
+      api.registerTopicFooterButton({
+        id: "share",
+        icon: "link",
+        priority: 999,
+        label: "topic.share.title",
+        title: "topic.share.help",
+        action: context => {
+          context.appEvents.trigger(
+            "share:url",
+            context.get("topic.shareUrl"),
+            $("#topic-footer-buttons")
+          );
+        },
+        dropdown: context => context.site.mobileView,
+        class: "share",
+        dependentKeys: ["topic.shareUrl", "topic.isPrivateMessage"],
```

Comment:
I am not crazy about the `dependentKeys` thing. Part of me thinks it should use the `@computed` helper on the methods that need it, but I realize that is super complicated.

A compromise might be to call it something closer to what it does. `refreshIfChanged` perhaps? `triggerRefresh`?

Reply:
Not sure, I like the fact that it does express the fact that an array is kind of expected. Also it's not exactly related to refresh, it's keys needed to build this button.

Reply:
Okay cool, keep it.